ERRORS.md

```diff
@@ -0,0 +1,54 @@
+# Errors in flyd
+
+## Ethos
++ It should not be the concern of `flyd` to handle exceptions for the user -- any `throw` should result in a hard failure.
++ Silent failures are bad (current way `flyd` handles Promise.reject)
++ Be as unopinionated in implementation as possible
++ Be functional in design
++ Be as backward compatible as possible with the current api
+
+## Concepts
++ The stream is of `events`
++ Each stream has a `left` and a `right` side (like an Either)
++ The right side is the domain objects
++ The left side is meta (in most cases errors)
++ By default the api operates on the `right` side
+
+## The Api
+`s` is a stream
+
+### Setting data s(...) is overloaded
++ `s(value)` is the default case takes a value makes it a right and pushes it down the stream
++ `s(promise)` if the promise resolves pushes a right, otherwise pushes a left
++ `s(either)` pushes down right or left based on either.left either.right
++ `s.left(value)` sets the stream to a left of `value`
+
+### Getting data
++ `s()` get the last right value or throws an exception if there is a left value
++ `s.left()` get the last left value or throws an exception if there is a right value
+
+### Checking stream state
++ `s.isLeft()` return boolean so you know what the stream contains
```

Comment:
this should also have a companion `s.isRight()` especially if `s()` can throw if it's not a right.